app/services/authorizer.rb

```diff
@@ -55,7 +55,7 @@ def find_collection(resource_class, options = {})
     all_filters = all_filters.reorder(nil).to_a # load all records, so #empty? does not call extra COUNT(*) query
     Foreman::Logging.logger('permissions').debug do
       all_filters.map do |f|
-        "filter with role_id: #{f.role_id} limited: #{f.limited?} search: #{f.search} taxonomy_search: #{f.taxonomy_search}"
+        "filter with role_id: #{f.role_id} limited: #{f.search_condition.present?} search: #{f.search} taxonomy_search: #{f.taxonomy_search}"
```

Comment:
The previous definition of `limited?` was slightly different https://github.com/theforeman/foreman/pull/10370/files#diff-0371bb2db296394504728985068e8db4f495b4639bec6ff517e63b857138dfdcL41

Reply:
True, but even in that PR, in one place `unlimited?` is replaced by `search_condition.blank?`. `limited?` was defined as `!unlimited?`, so `search_condition.present?` should be a 1:1 replacement. Am I mistaken?

Plus looking at the definition of search_condition:

```ruby
def search_condition
    searches = [search]
    searches << taxonomy_search
    searches.compact!
    searches.map! { |s| parenthesize(s) } if searches.size > 1
    searches.join(' and ')
end
```
It's just combining search and taxonomy_search, which is what the limited scope checked for.